src/openshift/component.ts

```diff
@@ -247,11 +248,15 @@ export class Component extends OpenShiftItem {
         let application: OpenShiftObject = context;
         if (!application) application = await Component.getOpenshiftData(context);
         if (!application) return null;
+        const delayer = new Delayer<string>(500);
         const repoURI = await vscode.window.showInputBox({
             prompt: 'Git repository URI',
             validateInput: (value: string) => {
-                if (!value.trim()) return 'Empty Git repository URL';
-                if (!validator.isURL(value)) return 'Invalid URL provided';
+                return delayer.trigger(async () => {
+                    if (!value.trim()) return 'Empty Git repository URL';
+                    console.log('validating!');
+                    if (!isGitRemote(value)) return 'No git repository exists. Please provide a valid git repository.';
```

Comment:
this will work only with git location added to PATH. We should use GIT API to check that without dependency on git executable.